### PR TITLE
Add migration hooks

### PR DESCRIPTION
**Pull request by apinnick**
https://issues.redhat.com/browse/MIG-226

CP 4.4, 4.5

### CODE REVIEW

**Review by sergiordlr — APPROVED** [review]
Looks good to me.

Thank you.

File: modules/migration-understanding-migration-hooks.adoc [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+* PreBackup: Before backup tasks are started on the source cluster
+* PostBackup: After backup tasks are complete on the source cluster
+* PreRestore: Before restore tasks are started on the target cluster
+* PostRestore: After restore tasks are complete on the target cluster
++
+You can assign one hook to each migration step, up to a maximum of four hooks for a single migration plan.
+
+A default `hook-runner` image, `quay.io/konveyor/hook-runner:latest`, is provided. This image is based on Ansible Runner and includes `python-openshift` for Ansible Kubernetes resources and an updated `oc` binary. You can also create your own hook image with additional Ansible modules or tools.
+
+The Ansible playbook is mounted on a hook container as a ConfigMap. The hook container runs as a Job on a cluster with a specified service account and namespace. The Job runs, even if the initial pod is evicted or killed, until it reaches the default `backoffLimit` (`6`) or successful completion.
```
Comment (ahardin-rh) [inline modules/migration-understanding-migration-hooks.adoc:27]: s/pod/Pod

**Comment by ahardin-rh** [comment]
Just one suggestion. LGTM!

**Review by ahardin-rh — APPROVED** [review]

**Pull request merged**